sycl/source/detail/global_handler.cpp

```diff
@@ -142,9 +142,11 @@ BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpReserved) {
   return TRUE; // Successful DLL_PROCESS_ATTACH.
 }
 #else
-// Setting minimal priority on destructor ensures it runs after all other global
-// destructors. Priorities 0-100 are reserved by the compiler.
-__attribute__((destructor(101))) static void syclUnload() { shutdown(); }
+// Setting low priority on destructor ensures it runs after all other global
+// destructors. Priorities 0-100 are reserved by the compiler. The priority
+// value 110 allows SYCL users to run their destructors after runtime library
+// deinitialization.
+__attribute__((destructor(110))) static void syclUnload() { shutdown(); }
```

Comment:
Should we document it somewhere what priority are for what? And how it interacts with no priority specified?

Reply:
I conducted a small experiment:
```c++
#include <iostream>

__attribute__((destructor(101))) static void syclUnload101() {
    std::cout << "101\n";
}

__attribute__((destructor(102))) static void syclUnload102() {
    std::cout << "102\n";
}

__attribute__((destructor(110))) static void syclUnload110() {
    std::cout << "110\n";
}

__attribute__((destructor(105))) static void syclUnload105() {
    std::cout << "105\n";
}

__attribute__((destructor)) static void syclUnload() {
    std::cout << "default\n";
}

int main() {
    std::cout << "Hi\n";
    return 0;
}
```

The result is the same for both GCC and Clang:
```
Hi
default
110
105
102
101
```